File: yht0827/1월/1월 17일/최대공약수와 최소공배수.java
class Solution {
	public int[] solution(int n, int m) {

		int a = Math.max(n, m);
		int b = Math.min(n, m);
		while (b != 0) {
			int r = a % b;
			a = b;
			b = r;
		}

		return new int[] {a, n * m / a};
	}
}

File: Eunleelee/2024-01/1월 17일/최대공약수와_최소공배수.py

def solution(n, m):
    answer = []
    li = [n, m]
    li.sort()
    max = 0
    for i in range(1, li[0] + 1):
        if n % i == 0 and m % i == 0:
            max = i
    
    j = 1
    while 1:
        if (li[1] * j) % li[0] == 0:
            min = int(li[1] * j)
            break
        j += 1
    
    answer = [max, min]
    return answer